Source/Diagnostics/BTDiagnostics.H

```diff
@@ -14,47 +14,88 @@ private:
     /** Read relevant parameters for BTD */
     void ReadParameters ();
     /** Flush m_mf_output and particles to file. */
-    // Currently this is an empty function. When OpenPMD/Plotfile is supported
-    // this function can be moved to the base class Diagnostics.
-    void Flush (int /*i_buffer*/) override {}
+    // Currently, a temporary customized output format for the buffer
+    // data is implemented and called in this function.
+    // After implementing OpenPMD and plotfile formats, this function
+    // can be defined and implemented in Diagnostics.H, such that,
+    // the function call to flush out buffer data for
+    // FullDiagnostics and BTDiagnostics is the same
+    void Flush (int i_buffer) override;
     /** whether to write output files at this time step
      *  The data is flushed when the buffer is full and/or
-     *   when the simulation ends or when forced.
+     *  when the simulation ends or when forced.
      * \param[in] step current time step
      * \param[in] force_flush if true, return true for any step
      *  The return bool is used to determine when to write buffer data out
      */
     bool DoDump (int step, int i_buffer, bool force_flush=false) override;
+    /** whether to compute the back-transformed data and store buffer in this timestep
+     *  The field-data is back-transformed from boosted-frame to lab-frame
+     *  at every time step within the PIC loop. Back-transformation is not performed
+     *  at initialization.
+     * \param[in] step current time step, return true for any step >= 0
+     * \param[in] force_flush if true, return true for any step
+     */
     bool DoComputeAndPack (int step, bool force_flush=false) override;
```

Comment:
Would sth like that work?
```suggestion
    bool DoComputeAndPack (int step, bool force_flush=false) {return step>=0;} override;
```